File: kagent/go/adk/pkg/a2a/consts.go
package a2a

import "google.golang.org/adk/server/adka2a"

const (
	StateKeySessionName = "session_name"
	StateKeySource      = "source"
)

// A2A DataPart metadata keys and type values.
const (
	A2ADataPartMetadataTypeKey              = "type"
	A2ADataPartMetadataIsLongRunningKey     = "is_long_running"
	A2ADataPartMetadataTypeFunctionCall     = "function_call"
	A2ADataPartMetadataTypeFunctionResponse = "function_response"
)

// DataPart map keys for GenAI-style function call / response content.
const (
	PartKeyName     = "name"
	PartKeyArgs     = "args"
	PartKeyResponse = "response"
	PartKeyID       = "id"
)

// HITL batch/rejection/ask-user constants.
const (
	KAgentHitlDecisionTypeBatch   = "batch"
	KAgentHitlDecisionsKey        = "decisions"
	KAgentHitlRejectionReasonsKey = "rejection_reasons"
	KAgentAskUserAnswersKey       = "ask_user_answers"
)

// ReadMetadataValue checks adk_<key> first, then kagent_<key>.
// Returns the value and true if found, or (nil, false).
func ReadMetadataValue(metadata map[string]any, key string) (any, bool) {
	if metadata == nil {
		return nil, false
	}
	adkKey := adka2a.ToA2AMetaKey(key)
	if v, ok := metadata[adkKey]; ok {
		return v, true
	}
	kagentKey := KAgentMetadataKeyPrefix + key
	if v, ok := metadata[kagentKey]; ok {
		return v, true
	}
	return nil, false
}

File: kagent/python/packages/kagent-core/src/kagent/core/a2a/_consts.py
# A2A DataPart metadata constants.
# These values MUST match the upstream google-adk definitions in
# google.adk.a2a.converters.part_converter. A sync-check test in
# kagent-adk verifies they stay in sync.
A2A_DATA_PART_METADATA_TYPE_KEY = "type"
A2A_DATA_PART_METADATA_IS_LONG_RUNNING_KEY = "is_long_running"
A2A_DATA_PART_METADATA_TYPE_FUNCTION_CALL = "function_call"
A2A_DATA_PART_METADATA_TYPE_FUNCTION_RESPONSE = "function_response"
A2A_DATA_PART_METADATA_TYPE_CODE_EXECUTION_RESULT = "code_execution_result"
A2A_DATA_PART_METADATA_TYPE_EXECUTABLE_CODE = "executable_code"

KAGENT_METADATA_KEY_PREFIX = "kagent_"
ADK_METADATA_KEY_PREFIX = "adk_"


def get_kagent_metadata_key(key: str) -> str:
    """Gets the A2A event metadata key for the given key.

    Args:
      key: The metadata key to prefix.

    Returns:
      The prefixed metadata key.

    Raises:
      ValueError: If key is empty or None.
    """
    if not key:
        raise ValueError("Metadata key cannot be empty or None")
    return f"{KAGENT_METADATA_KEY_PREFIX}{key}"


def read_metadata_value(metadata: dict | None, key: str, default=None):
    """Read a metadata value, checking ``adk_<key>`` first then ``kagent_<key>``.

    This allows interoperability with upstream ADK (which uses the ``adk_``
    prefix) while preserving backward-compatibility with kagent's own
    ``kagent_`` prefix.

    Args:
      metadata: The metadata dict to look up (may be ``None``).
      key: The unprefixed key name (e.g. ``"type"``).
      default: Value returned when the key is not found under either prefix.

    Returns:
      The value found under ``adk_<key>`` or ``kagent_<key>``, or *default*.

    Raises:
      ValueError: If *key* is empty or ``None``.
    """
    if not key:
        raise ValueError("Metadata key cannot be empty or None")
    if not metadata:
        return default
    adk_key = f"{ADK_METADATA_KEY_PREFIX}{key}"
    if adk_key in metadata:
        return metadata[adk_key]
    kagent_key = f"{KAGENT_METADATA_KEY_PREFIX}{key}"
    if kagent_key in metadata:
        return metadata[kagent_key]
    return default


KAGENT_HITL_DECISION_TYPE_KEY = "decision_type"
KAGENT_HITL_DECISION_TYPE_APPROVE = "approve"
KAGENT_HITL_DECISION_TYPE_REJECT = "reject"
KAGENT_HITL_DECISION_TYPE_BATCH = "batch"
KAGENT_HITL_DECISIONS_KEY = "decisions"
KAGENT_HITL_REJECTION_REASONS_KEY = "rejection_reasons"

KAGENT_ASK_USER_ANSWERS_KEY = "ask_user_answers"
